### ISSUE TITLE
A second text document opened on the same server/user will not save changes.

### ISSUE THREAD

**Issue by okennedy** [nextcloud/text#126]
**Describe the bug**
Only one document can be edited per server/user.  When a second document is created, even 24 hours later, changes to the second document are not saved.  

This behavior was observed on two separate Nextcloud servers.  In each case, the first document worked perfectly, while the second would not sync changes.  In both cases, the first document continues to work perfectly and changes to it continue to be saved correctly.

**To Reproduce**
1. Create a text document
2. Add text
3. Close the document
4. Create a second text document
5. Add text and wait for 30 seconds (note the asterisk in the upper-right-hand corner indicating unsaved changes does not go away)
6. Close the second text document
7. Re-open the second text document (Steps 6, 7 may need to be repeated a second or third time)
8. Open the first text document
9. Add more text (note the asterisk indicating unsaved changes _does_ go away)
10. Close the first text document and re-open it.

**Expected behavior**
Edits to both text documents should be saved and available when the document is re-opened or opened from another web browser.  

**Screenshots**
n/a.

**Client details:**
 - OS: Linux Mint 19
 - Browser: Firefox
 - Version: 67.0.4
 - Device: Desktop

<details>
<summary>Server details</summary>

**Social app version:** 0.2.6

**Operating system**: Linux Mint 19

**Web server:** Nginx

**Database:** Postgresql

**PHP version:** 7.2-FPM

**Nextcloud version:** 16.0.1

</details>

<details>
<summary>Logs</summary>

#### Nextcloud log (data/nextcloud.log)
```
Insert your Nextcloud log here
```

#### Browser log
```
Strict-Transport-Security: The site specified a header that did not include a ‘max-age’ directive. files
Strict-Transport-Security: The site specified a header that did not include a ‘max-age’ directive. merged-index.js
Strict-Transport-Security: The site specified a header that did not include a ‘max-age’ directive. activity-sidebar.js
Strict-Transport-Security: The site specified a header that did not include a ‘max-age’ directive. scripts-for-file-app.js
Strict-Transport-Security: The site specified a header that did not include a ‘max-age’ directive. merged-files.js
Strict-Transport-Security: The site specified a header that did not include a ‘max-age’ directive. styles.css
Strict-Transport-Security: The site specified a header that did not include a ‘max-age’ directive. files
The resource from “https://[WEBSERVER URL]/apps/files/” was blocked due to MIME type (“text/html”) mismatch (X-Content-Type-Options: nosniff). files
JQMIGRATE: Migrate is installed, version 1.4.1 main.js:278:531
OCA.Viewer initialized viewer.js:154:262
Strict-Transport-Security: The site specified a header that did not include a ‘max-age’ directive. notifications
New search handler registered search.js:67:12
The following handler is already registered 
Object { id: Getter & Setter, mimes: Getter & Setter, component: Getter & Setter, group: Getter & Setter, … }
viewer.js:132:5276
The escapeHTML library is deprecated! It will be removed in nextcloud 19. main.js:750:58
The select2 library is deprecated! It will be removed in nextcloud 19. main.js:750:58
Strict-Transport-Security: The site specified a header that did not include a ‘max-age’ directive. Nunito-Bold.woff2
Strict-Transport-Security: The site specified a header that did not include a ‘max-age’ directive. Nunito-Regular.woff2
Strict-Transport-Security: The site specified a header that did not include a ‘max-age’ directive. 32
The humanFileSize library is deprecated! It will be removed in nextcloud 19. main.js:750:58
Strict-Transport-Security: The site specified a header that did not include a ‘max-age’ directive. whatsnew
Strict-Transport-Security: The site specified a header that did not include a ‘max-age’ directive. settings
Strict-Transport-Security: The site specified a header that did not include a ‘max-age’ directive. files
querying Whats New data was successful: nocontent main.js:223:1268
undefined main.js:223:1328
Strict-Transport-Security: The site specified a header that did not include a ‘max-age’ directive. recommendations
Strict-Transport-Security: The site specified a header that did not include a ‘max-age’ directive. lastused
Strict-Transport-Security: The site specified a header that did not include a ‘max-age’ directive. config
Strict-Transport-Security: The site specified a header that did not include a ‘max-age’ directive. mounts
Strict-Transport-Security: The site specified a header that did not include a ‘max-age’ directive. getstoragestats.php
could not load recommendation preview 
error { target: img, isTrusted: true, srcElement: img, eventPhase: 0, bubbles: false, cancelable: false, returnValue: true, defaultPrevented: false, composed: false, timeStamp: 1715, … }
main.js:14:12601
could not load recommendation preview 
error { target: img, isTrusted: true, srcElement: img, eventPhase: 0, bubbles: false, cancelable: false, returnValue: true, defaultPrevented: false, composed: false, timeStamp: 1746, … }
main.js:14:12601
Strict-Transport-Security: The site specified a header that did not include a ‘max-age’ directive. 1
could not load recommendation preview 
error { target: img, isTrusted: true, srcElement: img, eventPhase: 0, bubbles: false, cancelable: false, returnValue: true, defaultPrevented: false, composed: false, timeStamp: 1847, … }
main.js:14:12601
The select2 library is deprecated! It will be removed in nextcloud 19. main.js:750:58
The humanFileSize library is deprecated! It will be removed in nextcloud 19. main.js:750:58
Strict-Transport-Security: The site specified a header that did not include a ‘max-age’ directive. getstoragestats.php
Strict-Transport-Security: The site specified a header that did not include a ‘max-age’ directive. preview
Strict-Transport-Security: The site specified a header that did not include a ‘max-age’ directive. create
The escapeHTML library is deprecated! It will be removed in nextcloud 19. main.js:750:58
Strict-Transport-Security: The site specified a header that did not include a ‘max-age’ directive. fetch
Strict-Transport-Security: The site specified a header that did not include a ‘max-age’ directive. sync
Failed to update steps in collaboration plugin RangeError: "Position 27 out of range"
    resolve https://[WEBSERVER URL]/apps/text/js/1.1db6cc360739149bc329.js:1
    resolveCached https://[WEBSERVER URL]/apps/text/js/1.1db6cc360739149bc329.js:1
    resolve https://[WEBSERVER URL]/apps/text/js/1.1db6cc360739149bc329.js:1
    replace https://[WEBSERVER URL]/apps/text/js/1.1db6cc360739149bc329.js:1
    fromReplace https://[WEBSERVER URL]/apps/text/js/1.1db6cc360739149bc329.js:1
    apply https://[WEBSERVER URL]/apps/text/js/1.1db6cc360739149bc329.js:1
    maybeStep https://[WEBSERVER URL]/apps/text/js/1.1db6cc360739149bc329.js:1
    step https://[WEBSERVER URL]/apps/text/js/1.1db6cc360739149bc329.js:1
    receiveTransaction https://[WEBSERVER URL]/apps/text/js/1.1db6cc360739149bc329.js:67
    update https://[WEBSERVER URL]/apps/text/js/1.1db6cc360739149bc329.js:67
    syncService https://[WEBSERVER URL]/apps/text/js/2.ee31d5b4cdf247f740b3.js:66
    value https://[WEBSERVER URL]/apps/text/js/2.ee31d5b4cdf247f740b3.js:45
    value https://[WEBSERVER URL]/apps/text/js/2.ee31d5b4cdf247f740b3.js:45
    value https://[WEBSERVER URL]/apps/text/js/2.ee31d5b4cdf247f740b3.js:45
    value https://[WEBSERVER URL]/apps/text/js/2.ee31d5b4cdf247f740b3.js:23
    o https://[WEBSERVER URL]/core/js/dist/main.js?v=cf3b3a26-11:230
    H https://[WEBSERVER URL]/core/js/dist/main.js?v=cf3b3a26-11:230
    u https://[WEBSERVER URL]/core/js/dist/main.js?v=cf3b3a26-11:235
2.ee31d5b4cdf247f740b3.js:66:5394
receivedSteps newVersion 0 2.ee31d5b4cdf247f740b3.js:45:2706
The escapeHTML library is deprecated! It will be removed in nextcloud 19. main.js:750:58
Strict-Transport-Security: The site specified a header that did not include a ‘max-age’ directive. sync
The escapeHTML library is deprecated! It will be removed in nextcloud 19. main.js:750:58
failed to apply steps due to collission, retrying 2.ee31d5b4cdf247f740b3.js:23:3021
Strict-Transport-Security: The site specified a header that did not include a ‘max-age’ directive. sync
The escapeHTML library is deprecated! It will be removed in nextcloud 19. main.js:750:58
Strict-Transport-Security: The site specified a header that did not include a ‘max-age’ directive. sync
The escapeHTML library is deprecated! It will be removed in nextcloud 19. main.js:750:58
Source map error: request failed with status 404
Resource URL: https://[WEBSERVER URL]/js/spreed/merged-files.js?v=cf3b3a26-11
Source Map URL: backbone.marionette.min.js.map
Source map error: TypeError: NetworkError when attempting to fetch resource.
Resource URL: https://[WEBSERVER URL]/apps/text/js/1.1db6cc360739149bc329.js
Source Map URL: 1.1db6cc360739149bc329.js.map
Source map error: TypeError: NetworkError when attempting to fetch resource.
Resource URL: https://[WEBSERVER URL]/apps/text/js/2.ee31d5b4cdf247f740b3.js
Source Map URL: 2.ee31d5b4cdf247f740b3.js.map
Strict-Transport-Security: The site specified a header that did not include a ‘max-age’ directive. sync
The escapeHTML library is deprecated! It will be removed in nextcloud 19. main.js:750:58
Source map error: TypeError: NetworkError when attempting to fetch resource.
Resource URL: https://[WEBSERVER URL]/apps/viewer/js/viewer.js?v=cf3b3a26-11
Source Map URL: viewer.js.map
Source map error: TypeError: NetworkError when attempting to fetch resource.
Resource URL: https://[WEBSERVER URL]/apps/recommendations/js/main.js?v=cf3b3a26-11
Source Map URL: main.js.map
Source map error: TypeError: NetworkError when attempting to fetch resource.
Resource URL: https://[WEBSERVER URL]/core/js/dist/main.js?v=cf3b3a26-11
Source Map URL: main.js.map
Strict-Transport-Security: The site specified a header that did not include a ‘max-age’ directive. sync
The escapeHTML library is deprecated! It will be removed in nextcloud 19. main.js:750:58
Strict-Transport-Security: The site specified a header that did not include a ‘max-age’ directive. notifications
Strict-Transport-Security: The site specified a header that did not include a ‘max-age’ directive. sync
The escapeHTML library is deprecated! It will be removed in nextcloud 19. main.js:750:58
```

</details>

**Comment by juliushaertl** [comment]
Unfortunately I could not reproduce this. I assume you just typed some text when testing? Can you maybe share one of the broken files as a public link with edit permissions with me (email in my github profile)

**Comment by okennedy** [comment]
I just sent a link to the email in your profile.

**Comment by juliushaertl** [comment]
Should be fixed with #148.
